Integration of Display Manager (gdm/xdm/etc.) to X server communication pipe
with core server code - see sun-src/os/dtlogin.c for the main implementation.

Not suitable for upstream

--- xorg-server-21.1.20/configure.ac
+++ xorg-server-21.1.20/configure.ac
@@ -1659,6 +1659,7 @@ #
 XSERVER_CFLAGS="${XSERVER_CFLAGS} ${XSERVERCFLAGS_CFLAGS}"
 XSERVER_LIBS="$DIX_LIB $MI_LIB $OS_LIB"
 XSERVER_SYS_LIBS="${XSERVERLIBS_LIBS} ${SYS_LIBS} ${LIBS}"
+XSERVER_SYS_LIBS="${XSERVER_SYS_LIBS} -lproject"
 AC_SUBST([XSERVER_LIBS])
 AC_SUBST([XSERVER_SYS_LIBS])
 
--- xorg-server-21.1.20/dix/main.c
+++ xorg-server-21.1.20/dix/main.c
@@ -117,6 +117,11 @@ #include <X11/extensions/dpmsconst.h>
 #include "dpmsproc.h"
 #endif
 
+#ifdef SUNSOFT
+extern void DtloginInit(void);
+extern void DtloginCloseDown(void);
+#endif /* SUNSOFT */
+
 extern void Dispatch(void);
 
 CallbackListPtr RootWindowFinalizeCallback = NULL;
@@ -149,6 +154,13 @@ ScreenSaverAllowExposures = defaultScree
         InitBlockAndWakeupHandlers();
         /* Perform any operating system dependent initializations you'd like */
         OsInit();
+
+#ifdef SUNSOFT
+        /* Create pipe for dtlogin authentication info before we tell dtlogin
+           we're done and ready for it to run. */
+        DtloginInit ();
+#endif
+
         if (serverGeneration == 1) {
             CreateWellKnownSockets();
             for (i = 1; i < LimitClients; i++)
@@ -276,6 +288,11 @@ Dispatch();
 
         UnrefCursor(rootCursor);
 
+#ifdef SUNSOFT
+        /* Return to root privs before calling the rest of close down */
+        DtloginCloseDown ();
+#endif
+
         UndisplayDevices();
         DisableAllDevices();
 
--- xorg-server-21.1.20/os/Makefile.am
+++ xorg-server-21.1.20/os/Makefile.am
@@ -46,6 +46,9 @@ if XDMCP
 libos_la_SOURCES += $(XDMCP_SRCS)
 endif
 
+libos_la_SOURCES += dtlogin.c
+libos_la_LIBADD += -lproject
+
 if HAVE_LIBUNWIND
 AM_CFLAGS += $(LIBUNWIND_CFLAGS)
 libos_la_LIBADD += $(LIBUNWIND_LIBS)
--- xorg-server-21.1.20/os/auth.c
+++ xorg-server-21.1.20/os/auth.c
@@ -283,6 +283,19 @@ }
     return 0;
 }
 
+#ifdef SUNSOFT
+/* This function is called from dtlogin.c
+ * This is added to do "chmod authorization_file" since
+ * this file is owned by root and we need to change this
+ * to user logged on.
+ */
+const char *
+GetAuthFilename(void)
+{
+    return (authorization_file);
+}
+#endif
+
 #ifdef XCSECURITY
 
 XID
